sdk/Sdk.Generators/FunctionMetadataProviderGenerator/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -672,6 +672,24 @@ private bool TryGetAttributeProperties(AttributeData attributeData, Location? at
                 return true;
             }
 
+            private static bool IsNamedArgumentValid(TypedConstant? namedArgument)
```

Comment:
Should we name this to be specific to the array check? What makes this named argument valid?

We should also move this function to the extension class maybe?

https://github.com/Azure/azure-functions-dotnet-worker/blob/main/sdk/Sdk.Analyzers/Extensions/TypeSymbolExtensions.cs

Reply:
Named arguments are valid if they have a non-null value or if they are an array type (both checks are in this method) - could move this to an extension class but it's specific to this scenario.

"Valid" in this case means "we can/should write this to the function metadata". For example, with this [input binding](https://github.com/Azure/azure-functions-dotnet-worker/blob/main/extensions/Worker.Extensions.CosmosDB/src/CosmosDBInputAttribute.cs) which has nullable properties, even if you add a prop explicitly as a named argument with a null value in the attribute, it won't be written to the metadata. Validated that this is legacy generator behavior and emulating it here.